source/agora/consensus/EnrollmentManager.d

```diff
@@ -1142,3 +1158,62 @@ unittest
     man.getEnrollments(enrolls, Height(validator_cycle + 11));
     assert(enrolls.length == 0);
 }
+
+/// Test for restoring enrollment data and pre-image information for a node
+unittest
+{
+    import agora.common.Amount;
+    import agora.consensus.data.Transaction;
+    import agora.consensus.Genesis;
+    import std.conv;
+
+    scope utxo_set = new TestUTXOSet;
+    auto gen_key_pair = WK.Keys.Genesis;
+    auto key_pair = KeyPair.random();
+
+    foreach (uint idx; 0 .. 7)
+    {
+        auto tx = TxBuilder(GenesisTransaction, idx).
+            draw(Amount(Amount.MinFreezeAmount), [gen_key_pair.address]).
+            sign(TxType.Freeze);
+        utxo_set.put(tx);
+    }
```

Comment:
This is invalid, the UTXOs will double spend, which only works because we don't have a ledger instance.